minimaxPlayer.py

```diff
@@ -0,0 +1,126 @@
+import numpy as np
+import math
+from mancala import Player, MancalaBoard
+
+class MinimaxPlayer(Player):
+
+    def __init__(self, isPlayer1) -> None:
+        super().__init__(isPlayer1)
+        if isPlayer1:
+            self.getNextMove = self.getNextMove1
+        else:
+            self.getNextMove = self.getNextMove2
+
+    def getNextMove1(self, boardState) -> int:
+
+        board = MancalaBoard()
+        board.board = np.copy(boardState)
+        bestMove = self.minimax(board, 0, True, True)
+        return bestMove[0]
+
+    def getNextMove2(self, boardState) -> int:
+        board = MancalaBoard()
+        board.board = np.copy(boardState)
+        bestMove = self.minimax(board, 0, True, False)      
+        return bestMove[0]
+
+    # function -> minimax 
+    #   returns a tuple (max pit index, max relative sum of pits)
+    # parameters:
+    # boardState -> board object
+    # depth -> current depth (increasing)
+    # player -> true (maximizer) | false (minimizer)
+    # player1 -> true (player1) | false (player2)
+
+    def minimax(self, boardState, depth, player, player1):
+       
+        if player1 == True:
```

Comment:
Instead of a big if statement, just move each branch into each function.

Reply:
Gotcha, taken care of and repushed